spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,122 @@
+require_relative "spec_helper"
+
+describe "ReservationManager" do
+  let(:manager_new) do
+    Hotel::ReservationManager.new
+  end
+
+  let(:reservation_generator) do
+    # Reservation 1
+    start_date = Time.parse("2019-03-11 14:08:45 -0700")
+    end_date = Time.parse("2019-03-15 14:08:45 -0700")
+    room = 3
+    manager_new.create_reservation(start_date, end_date, room)
+    # Reservation 2
+    start_date = Time.parse("2019-03-20 14:08:45 -0700")
+    end_date = Time.parse("2019-03-22 14:08:45 -0700")
+    room = 20
+    manager_new.create_reservation(start_date, end_date, room)
+    # Reservation 3
+    start_date = Time.parse("2019-02-27 14:08:45 -0700")
+    end_date = Time.parse("2019-02-28 14:08:45 -0700")
+    room = 12
+    manager_new.create_reservation(start_date, end_date, room)
+    # Reservation 4
+    start_date = Time.parse("2019-03-19 14:08:45 -0700")
+    end_date = Time.parse("2019-03-21 14:08:45 -0700")
+    room = 6
+    manager_new.create_reservation(start_date, end_date, room)
+  end
+
+  it "creates a reservation" do
+    expect(reservation_generator).must_be_instance_of Hotel::Reservation
+  end
+
+  it "raises ArgumentError when date range is invalid" do
+    id = 1
+    start_time = Time.parse("2019-03-19 14:08:45 -0700")
+    end_time = Time.parse("2019-03-15 14:08:45 -0700")
+    room_number = 3
+
+    expect { Hotel::ReservationManager.new(start_time, end_time, room_number) }.must_raise ArgumentError
+  end
+
+  it "creates a reservation for an available room" do
+    reservation_generator
+    reservation = manager_new.create_reservation(Time.parse("2019-03-21 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 6)
+    expect(reservation).must_be_instance_of Hotel::Reservation
+  end
+
+  it "raises ArgumentError when a room chosen is not existent" do
+    expect { manager_new.create_reservation(Time.parse("2019-03-21 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 24) }.must_raise ArgumentError
+  end
+
+  it "raises ArgumentError when a room chosen is not available" do
+    reservation_generator
+
+    expect { manager_new.create_reservation(Time.parse("2019-03-19 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 6) }.must_raise ArgumentError
+  end
+
+  it "raises ArgumentError when range of dates are invalid" do
```

Comment:
You should also verify that it works when one reservation starts on the same day another ends.